src/components/related/RelatedProductsList.jsx

```diff
@@ -44,21 +46,97 @@ class RelatedProductsList extends React.Component {
+
+    const numberOfCardsToShow = Math.floor(visibleWidth / cardWidth);
+    let indexesArray = [0];
+
+    for (let i = 1; i < numberOfCardsToShow; i++) {
+      indexesArray.push(i);
+    }
+
+    this.setState({
+      indexesToShow: indexesArray
+    });
+  }
+
+  handleLeftArrowClick() {
+    const { indexesToShow, relatedIds } = this.state;
+    let { showLeftArrow, showRightArrow } = this.state;
+
+    if (indexesToShow[0] - 1 === 0) {
+      showLeftArrow = false;
+    }
+
+    if (indexesToShow[indexesToShow.length - 1] < relatedIds.length) {
+      showRightArrow = true;
+    }
+
+    this.setState({
+      showLeftArrow: showLeftArrow,
+      showRightArrow: showRightArrow,
+      indexesToShow: indexesToShow.map(index => { return index - 1; })
+    });
+  }
+
+  handleRightArrowClick() {
+    const { indexesToShow, relatedIds } = this.state;
+    let { showLeftArrow, showRightArrow } = this.state;
+
+    if (indexesToShow[0] + 1 > 0) {
+      showLeftArrow = true;
+    }
+
+    if (indexesToShow[indexesToShow.length - 1] + 2 === relatedIds.length) {
+      showRightArrow = false;
+    }
+
+    this.setState({
+      showLeftArrow: showLeftArrow,
+      showRightArrow: showRightArrow,
+      indexesToShow: indexesToShow.map(index => { return index + 1; })
+    });
+
+  }
+
   render() {
     const { productId, productCardId, updateAppProductId,
-      currentList, removeItemFromOutfit} = this.props;
+      currentList, removeItemFromOutfit } = this.props;
+    const { showLeftArrow, indexesToShow, relatedIds } = this.state;
+    let { showRightArrow } = this.state;
+
+    if (relatedIds.length <= indexesToShow.length) {
+      showRightArrow = false;
+    }
 
     return (
       <>
         <h3>Related Products</h3>
-        <div className='product-card-list'>
-        {this.state.relatedIds.map((relatedId, i) => {
-          return <ProductCard key={i}
-            currentList={currentList}
-            productId={productId}
-            productCardId={relatedId}
-            updateAppProductId={updateAppProductId} />
-        })}
+        <div id='related-list'>
+          <div className='related-arrow'>
+            <img className={showLeftArrow ? 'related-left-arrow' : 'related-left-arrow hide'}
+              src='https://i.ibb.co/r0GN44X/image.png'
+              onClick={() => { this.handleLeftArrowClick() }}></img>
+          </div>
+          <div className='product-card-list'>
+            {this.state.relatedIds.map((relatedId, i) => {
+              return <ProductCard key={i}
+                hidden={!indexesToShow.includes(i)}
+                currentList={currentList}
+                productId={productId}
+                productCardId={relatedId}
+                updateAppProductId={updateAppProductId} />
+            })}
+          </div>
+          <div className='related-arrow'>
+            <img className={showRightArrow ? 'related-right-arrow' : 'related-right-arrow hide'}
```

Comment:
this is great! I didn't know we're able to do the conditional rendering inside of className.

Reply:
🧊 super cool